C++ Secret Messages English Translation (#270)
NuevoFoundation/workshops: content/english/secret-messages/activity-2.md

{{% /notice %}}

{{< notice note >}}
We can use `cout` to print words, numbers or variables. If we would to print more that one thing, we would have to add `<<` between each term:
Example: 
```
int hour = 13;
cout << "It is " << hour << " in the afternoon" << endl;
```
This will print:
```
It is 13 in the afternoon.
```
{{< /notice >}}

Next, we'd like the recipient of the message to be able to write his or her name to the console, (which we will be storing inside a variable), and the program to say "Hello" to him or her.

To input the name we will be using the following code:
```
string name;
cin >> name;
```
Let's dig into this line of code.

{{% notice tip %}}

### Console Input

Similar to how we can print to console using `cout`, we can input data into the console by using `cin`, and specifying the name of the variable in which the input will be stored.

In the previous example we are stating that whatever data we input, it will be stored inside the variable `name`.

That being said, if we input:
```
Queen
```
Then the variable `name` will store the string `"Queen"`.

{{% /notice%}}

Now it's time to begin with our program.

These are the steps to follow for this activity:

> 1. Declare a string variable named `name`

Reviewer: ```suggestion
1. Declare a string variable named `name`.
```

cout << "The number is: " << number << endl;
```
 
Console will print:

```
The number is: 10
```
{{% /notice %}}

{{< notice note >}}
We can use cout to print words, numbers or variables. If we would to print more that one thing, we would have to add `<<` between each term:
Example: 
```
int hour = 13;
cout << "It is" << hour << " in the afternoon" << endl;
```
This will print:
```
It is 13 in the afternoon.
```
{{< /notice >}}

The next thing we need is the ability to type into console the name of the recipient (which we will be storing inside a variable), and the program to say hello to him or her.

To input the name we will be using the following code:
```
string name;
cin >> name;
```
Lets dig into this line of code.

{{% notice tip %}}

### Console Input

As same as we can print to console using `cout`, we can input data through console by using `cin` specifying the name of the variable in which whatever input will be stored.

In the previous example we are stating that whatever data we input, it will be stored inside the variable **name**.

That being said, if we input:
```
Queen
```
Then the variable `name` will store the String "Queen".

{{% /notice%}}

It is time then!

> Lets begin with our program.

Reviewer: ```suggestion
Let's begin with our program.
```

These are the steps to follow for this activity:

> 1. Declare a String variable.
> 2. Use `cin` to input the data that will be stored in the variable.
> 3. Print to console "Hello " and the data stored in the variable.
> 4. Click Run. Type in the name of the recipient in the **console** tab.

Reviewer: ```suggestion
1. Declare a string variable named `name`
2. Use `cin` to request input from the user, and store the inputted data into the `name` variable.
3. Print to console "Hello (name)", where (name) should be replaced by the value of the `name` variable.
4. Click Run. Type in "Queen" as the name of the recipient in the **console** tab, and make sure you get "Hello Queen" as output from the console.
```

<iframe height="600px" width="100%" src="https://replit.com/@nuevofoundation/activity-2?lite=true#main.cpp" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

{{% notice warning %}}

### HELP!

> In case that after clicking the Run button, an error message in red shows, it means that there is something wrong with your code. Please ask for help to your instructor.

Reviewer: ```suggestion
If after clicking the Run button, an error message in red appears, it means that there is something wrong with your code. Please ask for help from your instructor.
```

> In the meantime you could check for:
> 
> 1. Do not use caps.
> 2. Make sure that you add a semicolon after each line of code. 
> 3. Remember that all Strings have to go between quotation marks.
> 4. For now, all code that write will have to go inside the brackets `{}` on the `int main` function.

Reviewer: ```suggestion
You should also check for the following
1. Ensure the spelling, capitalization, and symbols used match with the example given.
2. Make sure that you add a semicolon at the end of each line of code. 
3. Remember that all strings must be between quotation marks.
4. All the code that you write need to be within the brackets `{}` in the `int main` function.
```
